Build failure: units

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Try to build `units`.

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/318219149

### Relevant log output

```console
In file included from units.c:92:
units.h:57:12: error: conflicting types for 'strchr'; have 'char *(void)'
   57 |      char *strchr();
      |            ^~~~~~
In file included from /nix/store/inys8r1mg4mwlxnrwa3wqga3xmp1rmvi-readline-8.3p1-dev/include/readline/chardefs.h:35,
                 from /nix/store/inys8r1mg4mwlxnrwa3wqga3xmp1rmvi-readline-8.3p1-dev/include/readline/keymaps.h:35,
                 from /nix/store/inys8r1mg4mwlxnrwa3wqga3xmp1rmvi-readline-8.3p1-dev/include/readline/readline.h:37,
                 from units.c:84:
/nix/store/dj43clc5ff7jjnfmhbaj6q4q0h8kpfpm-glibc-2.40-66-dev/include/string.h:246:14: note: previous declaration of 'strchr' with type 'char *(const char *, int)'
  246 | extern char *strchr (const char *__s, int __c)
      |              ^~~~~~
units.c: In function 'hassubscript':
units.c:495:10: error: too many arguments to function 'strchr'; expected 0, have 2
  495 |     if (!strchr(digits, *ptr))
      |          ^~~~~~ ~~~~~~
units.h:57:12: note: declared here
   57 |      char *strchr();
      |            ^~~~~~
```

### Additional context

The log errors are hundreds of lines, but they're all the same complaints about `strchr`.

Probably due to #475479.

### System metadata

Fails on Hydra.

### Notify maintainers

No maintainers listed.  😢

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/un/units/package.nix
+++ b/pkgs/by-name/un/units/package.nix
@@ -21,6 +21,9 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-HlAsTt+s8gspKEcWxy5d21GklaI2XXsD55YElMSgyQI=";
   };
 
+  # Until upstream updates their code to work with GCC 15.
+  env.NIX_CFLAGS_COMPILE = "-std=gnu17";
+
   outputs = [
     "out"
     "info"
@@ -71,7 +74,7 @@ stdenv.mkDerivation (finalAttrs: {
     '';
     license = with lib.licenses; [ gpl3Plus ];
     mainProgram = "units";
-    maintainers = [ ];
+    maintainers = with lib.maintainers; [ galen ];
     platforms = lib.platforms.all;
   };
 })

